extensions/ql-vscode/src/model-editor/languages/python/index.ts

```diff
@@ -0,0 +1,207 @@
+import type { ModelsAsDataLanguage } from "../models-as-data";
+import { sharedExtensiblePredicates, sharedKinds } from "../shared";
+import { Mode } from "../../shared/mode";
+import type { MethodArgument } from "../../method";
+import { EndpointType, getArgumentsList } from "../../method";
+import {
+  parsePythonAccessPath,
+  pythonEndpointType,
+  pythonMethodPath,
+  pythonMethodSignature,
+  pythonPath,
+} from "./access-paths";
+
+export const python: ModelsAsDataLanguage = {
+  availableModes: [Mode.Framework],
+  createMethodSignature: ({ typeName, methodName }) =>
+    `${typeName}#${methodName}`,
+  endpointTypeForEndpoint: (method) => pythonEndpointType(method),
+  predicates: {
+    source: {
+      extensiblePredicate: sharedExtensiblePredicates.source,
+      supportedKinds: sharedKinds.source,
+      supportedEndpointTypes: [EndpointType.Method, EndpointType.Function],
+      // extensible predicate sourceModel(
+      //   string type, string path, string kind
+      // );
```

Comment:
Not necessarily related to this PR (since the other languages do the same), but I was wondering what all these commented-out extensible predicates are for 🤔 Are we waiting on a new feature before these can be used?

Reply:
These are basically just reminders for what the extensible predicates look like since this is very useful to know for knowing what `row[0]`, `row[1]`, etc. refer to. In this case, this is referring to [this definition](https://github.com/github/codeql/blob/fb2d36ddac8b719f7679eb458383d61706507a01/python/ql/lib/semmle/python/frameworks/data/internal/ApiGraphModelsExtensions.qll#L11).